compiler/resolution/functionResolution.cpp

```diff
@@ -4931,14 +4931,56 @@ createFunctionAsValue(CallExpr *call) {
 
   FnSymbol* captured_fn = visibleFns.head();
 
-  //Check to see if we've already cached the capture somewhere
-  if (functionCaptureMap.find(captured_fn) != functionCaptureMap.end()) {
-    return new CallExpr(functionCaptureMap[captured_fn]);
+  //
+  // Is this a case where we need a C function pointer or a Chapel
+  // first-class function / closure?
+  //
+  bool cFunctionPtr = false;
+
+  //
+  // If the capture call has two arguments, it implies that we added a
+  // sentinel in scopeResolve.cpp because we detected a call to
+  // c_fnPtrTo()).  So this should just be a simple function pointer,
+  // not a Chapel first-class function.
+  //
+  if (call->numActuals() == 2) {
```

Comment:
This seems like a bit of a magical way of determining whether we're in a capture call.  Does the sentinel argument have a specific name we can key off of?  Or something that is more unique than "it has two arguments"?